### PR TITLE
Versioned search implementation

### PR DESCRIPTION
**Pull request by riccardozambito**
- Versioned search implemented
- Lint fixes

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -37,119 +42,182 @@ module.exports = function (context, options) {
...
+          ),
+          "next",
+        ];
+
+        let version = "all";
+
+        if (docVersions && docVersions.length) {
+          version = docVersions[0];
+
+          const matchedVersion = d.url.match(/docs\/(\d+\.x\.x)/);
```
Comment (davidebianchi) [inline src/index.js:114]: We should comment this line, and write something about it in the README.
To match this, the url should be with the version in url with the shape of `number.x.x`. It is not possible to pass the regex in some options?

In the default case (without version or next), why it is taken the `all` version?

File: src/index.js [hunk 2]
```diff
@@ -1,31 +1,36 @@
-const fs = require('fs')
-const os = require('os')
-const path = require('path')
-const lunr = require('lunr')
-const { Worker } = require('worker_threads')
-const Guage = require('gauge')
+const fs = require("fs");
```
Comment (davidebianchi) [inline src/index.js:1]: Could you remove all the unnecessary changes from the js files?

**Review by davidebianchi — CHANGES_REQUESTED** [review]

**Comment by riccardozambito** [comment]
Closed.
New PR -> #2
